packages/cozy-client/src/helpers.spec.js

```diff
@@ -109,3 +109,33 @@ describe('dehydrate', () => {
     )
   })
 })
+
+describe('generateWebLink', () => {
+  it('should generate the right link to a flat cozy', () => {
+    const sharecode = 'sharingIsCaring'
+    const username = 'alice'
+
+    expect(
+      generateWebLink({
+        cozyUrl: 'http://alice.cozy.tools',
+        searchParams: [['sharecode', sharecode], ['username', username]],
+        nativePath: '/n/4',
+        slug: 'notes',
+        subDomainType: 'flat'
+      })
+    ).toEqual(
+      `http://alice-notes.cozy.tools/?sharecode=${sharecode}&username=${username}#/n/4`
+    )
+  })
+
+  it('should generate the right link to a nested cozy', () => {
+    expect(
+      generateWebLink({
+        cozyUrl: 'https://alice.cozy.tools/public/',
```

Comment:
I don't like to put the "public" path to the cozyURL. CozyURL should only be the base domain.

Reply:
I thought of adding a `publicView` boolean option but then I thought that using the `cozyUrl` option would be more generic.

Reply:
I don't think so. `https://alice.cozy.tools/public/` is not a valid URL and should not be put to a `cozyURL` var. 

Today I would prefer to have a `path` attribute since some apps can have : `foo.cozy.tools/public?shareCode` and others `foo.cozy.tools/public/?shareCode`

Reply:
Why is it not valid? I'm ok with adding a `path` parameter though. Would you rather have a `/` at the end or not?

Reply:
By not valid I mean that the URL doesn't exist. 
```
>curl -o /dev/null --silent --head --write-out '%{http_code}\n' https://crash-notes.mycozy.cloud/public/
200

> curl -o /dev/null --silent --head --write-out '%{http_code}\n' https://crash.mycozy.cloud/public/
404
```

We should follow the URL spec and use pathname in fact : https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname